### ISSUE TITLE
[merge] MarkAttachClassDef/MarkFilteringSets

### ISSUE THREAD

**Issue by khaledhosny** [fonttools/fonttools#1951]
Currently the classes/sets from different fonts will be merged together, so if font A and font B both has class/set 1, the merged font will also have 1 then contains all the glyphs in both classes/sets, instead of adding another class/set for the second font.

This seems to be work, but not ideal. There is a TODO in `_preMerge()` and `_postMerge()` to handle it, and I’m willing to give it a try if there are some pointers how this should be handled.

**Comment by khaledhosny** [comment]
CC @behdad

**Comment by behdad** [comment]
Yeah I can give you some starting points for that.  That said, I'm also thinking about a new modules (possibly called `processor` or `dispatch`) that will improve on and centralize the programming scheme used in the `merge`, `subset`, and `varLib` modules.  That will simplify a lot of things.   Give me a couple of days.

**Comment by behdad** [comment]
> Give me a couple of days.

To write out my thoughts, not implement it. :))

**Comment by khaledhosny** [comment]
OK :)

**Comment by behdad** [comment]
At least MarkFilteringSets were done in b8615f9de653a170b9e4e2ba7ec1d0fd3539fd65

MarkAttachClassDefs are bigger piece of work.

**Comment by behdad** [comment]
> MarkAttachClassDefs are bigger piece of work.

And can very easily overflow the allowed 255...

In fact, that's the only place that is currently broken in, eg. a merged Noto.
